cmake/FindKOKKOS.cmake

```diff
@@ -1,12 +1,21 @@
 # - Find kokkos
-# Creates a Kokkos::kokkos imported target
+# Creates a PkgConfig::KOKKOS imported target
 #
 #  KOKKOS_INCLUDE_DIRS  - where to find kokkos.h, etc.
 #  KOKKOS_LIBRARIES     - List of libraries when using kokkos.
 #  KOKKOS_FOUND         - True if kokkos found.
 #  KOKKOS_SETTINGS_DIR - path to kokkos_generated_settings.cmake
 #
 
+find_package(PkgConfig)
```

Comment:
@rfbird @sslattery should we require pkg-config and drop all the code below?

Reply:
Will that work across platforms? I'll admit I've never used it so not sure if it will work on Mac or the HPC machines

Reply:
I just tested this on a bunch of machines and it seems the `IMPORTED_TARGET` capability isn't full there in CMake-3.9, so for the sake of not having to bump the minimal CMake version, I will rewrite that a bit.